AutoComplete Feature Test.

K-cloud25 opened the pull request:
Files Created : AutoComplete.js
AutoComplete Component.
Based on current input check with list fo branches passed as props from parent. Based on closest match suggestion is shown in terminal. useInput hook utilized to auto update value.

Updated Files : DeleteBranch.js
Replaced text Component with AutoComplete Component

K-cloud25 committed:
AutoComplete Feature Test.

Files Created : AutoComplete.js
AutoComplete Component.
Based on current input check with list fo branches passed as props from parent. Based on closest match suggestion is shown in terminal. useInput hook utilized to auto update value.

Updated Files : DeleteBranch.js
Replaced text Component with AutoComplete Component

H3RSKO commented on components/AutoComplete.js:
```diff
@@ -0,0 +1,37 @@
+const { Box, Text, useInput } = require("ink");
+
+const AutoComplete = ({valueControl={val, setValue}, baseList, handleSubmit}) => {
+    const [suggestion, setSuggestion] = React.useState([]);
```
`suggestion` defaults as an array but is treated as a string later.

```suggestion
    const [suggestion, setSuggestion] = React.useState('');
```

H3RSKO commented on components/DeleteBranch.js:
```diff
@@ -1,9 +1,11 @@
-const TextInput = require("ink-text-input").default;
+const AutoComplete = importJsx("./AutoComplete");
 
+const words = ["apple", "application", "banana", "band", "cat", "caterpillar"];
```
```suggestion
```

H3RSKO commented on components/AutoComplete.js:
```diff
@@ -0,0 +1,37 @@
+    const updateSuggestion = (input) => {
+		valueControl.setValue(input);
+		if (input.length > 0) {
+		  const match = baseList.find(w => w.trim().startsWith(input.toLowerCase()));
```
We should make sure **both** input and the item from the baseList are lowercase.

```suggestion
          const match = baseList.find(w => w.trim().toLowerCase().startsWith(input.toLowerCase()));
```

H3RSKO reviewed (changes requested):
Posted some comments, but the core functionality looks good! Maybe you can also add the AutoComplete functionality to the Checkout Branch as well?

K-cloud25 committed:
Update components/AutoComplete.js

Co-authored-by: Yosef Herskovitz <34112131+H3RSKO@users.noreply.github.com>

K-cloud25 committed:
Update components/DeleteBranch.js

Co-authored-by: Yosef Herskovitz <34112131+H3RSKO@users.noreply.github.com>

K-cloud25 committed:
Update components/AutoComplete.js

Co-authored-by: Yosef Herskovitz <34112131+H3RSKO@users.noreply.github.com>

K-cloud25 committed:
Added Autocompleteto CheckoutBranch page.

Removed single quotes (') from the format for git call for branches. Ensures user does not have to type out the single quote.

Seperated the list of branchs with '|' to make the list more readable.

K-cloud25 commented:
Hi @H3RSKO  Added the requested comments to code

Added some updates to the commit as well. Please review them and let me know if any changes are required.

K-cloud25 commented:
On a side not I would like to improve the functionality of the AutoComplete component. 

If I could generate a list of suggestions instead of a singular option and one the press of the up or down key iterate through the generated list. The list could be sorted using levenshtein distance compared with the user input. 

Let me know if I should work on this. I would like to work on it.

K-cloud25 commented:
Hi @H3RSKO  could you merge the request if all is correct?
